packages/create-expo-module/src/create-expo-module.ts

```diff
@@ -210,14 +210,44 @@ async function getNpmTarballUrl(packageName: string, version: string = 'latest')
   return stdout.trim();
 }
 
+/**
+ * Gets expo SDK version from the local package.json.
+ */
+async function getLocalSdkVersion(): Promise<string | null> {
+  const packageJsonPath = await findUp('package.json', { cwd: CWD });
+  if (!packageJsonPath) {
+    return null;
+  }
+  const packageJson = await fs.readJSON(packageJsonPath);
+  return packageJson.dependencies?.expo ?? null;
+}
+
+/**
+ * Selects correct version of the template based on the SDK version for local modules and EXPO_BETA flag.
+ */
+async function getTemplateVersion(targetDir: string, isLocal) {
+  if (EXPO_BETA) {
+    return 'next';
+  }
+  if (!isLocal) {
+    return 'latest';
+  }
+  const sdkVersion = await getLocalSdkVersion();
+
+  const npmMajorOnlyRegex = /([0-9]+)\..*/m;
+  return sdkVersion ? sdkVersion.match(npmMajorOnlyRegex)?.[1] : 'latest';
```

Comment:
If my previous suggestion works, it's better to do `sdkVersion.split('.')` and get the first item.

Other than that, I think we'll go with `sdk-X` tags (as the branch name from which we do the release). I'm not even sure if it's possible to use digits-only tags as it would conflict with versions.
See this note 👇 from [npm docs](https://docs.npmjs.com/adding-dist-tags-to-packages).

> We recommend avoiding dist-tags that start with a number or the letter "v"

Reply:
I thought we want to release templates with versions matching SDK versions, not with SDK tags, so it was supposed to be a version – we can do tags instead

Reply:
instead of `sdkVersion.split('.')` i'll use `semver.major()`

Reply:
In general I would say yes, but to be honest I'm reluctant to add another dependency just for this simple thing 😅

Reply:
I think it's already used as a transitive dependency and in other packages of ours, so I'd lean towards using that?